src/hooks/useModal/useModal.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback, useState } from 'react'
+
+import { ModalAPI } from './useModal.types'
+
+export const useModal = (): ModalAPI => {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+
+  const openModal = useCallback(() => setIsModalVisible(true), [setIsModalVisible])
+  const closeModal = useCallback(() => setIsModalVisible(false), [setIsModalVisible])
+  const changeModalStatus = useCallback(() => setIsModalVisible(!isModalVisible), [setIsModalVisible, isModalVisible])
```

Comment:
Questo è un uso scorretto del setState. 
Il set state è asincrono, se bindi così il vecchio visible potrebbe non avere il valore che ti aspetti.
Per avere la certezza di sapere quale l'attuale valore nel tuo state devi usare questa sintassi

```suggestion
  const changeModalStatus = useCallback(() => setIsModalVisible(prevState => !prevState), [])
```